.github/workflows/main.yml

```diff
@@ -40,25 +64,242 @@ jobs:
+
+      - name: Setup Node for Newman
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install Newman
+        run: npm i -g newman newman-reporter-htmlextra
+
+      - name: Inject sensitive variables (optional)
+        if: always()
+        shell: bash
+        run: |
+          # If you keep sensitive secrets out of Postman, you can override here.
+          # Example: pass API tokens as Newman env-vars (picked in collection via {{apiToken}})
+          case "$TARGET_ENV" in
+            dev)  TOKEN="${{ secrets.DEV_API_TOKEN }}" ;;
+            qa)   TOKEN="${{ secrets.QA_API_TOKEN }}" ;;
+            *)    TOKEN="" ;;
+          esac
+          echo "API_TOKEN=$TOKEN" >> "$GITHUB_ENV"
+
+      - name: Run all Postman collections with Newman
+        shell: bash
+        run: |
+          set -e
+          mkdir -p newman
+          ENV_FILE="${{ steps.fetch-postman.outputs.env_file }}"
+          GLOBALS_FLAG="${{ steps.fetch-postman.outputs.globals_flag }}"
+          # Optionally, run only a "Smoke" folder from each collection first:
+          # FOLDER_FLAG='--folder "Smoke"'
+          FOLDER_FLAG=''
+          FAILED_COLLECTIONS=0
+          TOTAL_COLLECTIONS=0
+          for COL in postman/*.collection.json; do
+            BASENAME=$(basename "$COL" .collection.json)
+            TOTAL_COLLECTIONS=$((TOTAL_COLLECTIONS + 1))
+            echo "========================================"
+            echo "Running collection: $BASENAME"
+            echo "========================================"
+            if newman run "$COL" \
+              -e "$ENV_FILE" \
+              $GLOBALS_FLAG \
+              $FOLDER_FLAG \
+              --reporters cli,htmlextra,junit \
+              --reporter-htmlextra-export "newman/${BASENAME}.html" \
+              --reporter-junit-export "newman/${BASENAME}.xml" \
+              --timeout-request 120000 \
+              --delay-request 50 \
+              --env-var "apiToken=${API_TOKEN}"; then
+              echo "✅ Collection passed: $BASENAME"
+            else
+              echo "❌ Collection failed: $BASENAME"
+              FAILED_COLLECTIONS=$((FAILED_COLLECTIONS + 1))
+            fi
+            echo ""
+          done
+          echo "========================================"
+          echo "Test Summary"
+          echo "========================================"
+          echo "Total collections: $TOTAL_COLLECTIONS"
+          echo "Passed: $((TOTAL_COLLECTIONS - FAILED_COLLECTIONS))"
+          echo "Failed: $FAILED_COLLECTIONS"
+          if [ "$FAILED_COLLECTIONS" -gt 0 ]; then
+            echo "❌ Some collections failed"
+            exit 1
+          else
+            echo "✅ All collections passed!"
+          fi
+
+      - name: Upload test reports
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: newman-reports-${{ env.TARGET_ENV }}
```

Comment:
Should we include the commit hash in the name here to differentiate what failed/fixed a given test run?